crates/wallet/src/wallet/mod.rs

```diff
@@ -2299,6 +2228,16 @@ impl Wallet {
         self.indexed_graph.graph()
     }
 
+    /// Get a mutable reference to the inner [`TxGraph`].
+    ///
+    /// **WARNING**: It's not recommended to make changes manually to the transaction
```

Comment:
❓ Is this only being exposed for testing purposes? If so can we avoid move the tests that need it into a test module within this crate so it can be made `pub(crate)` instead? I hate to leave this foot-gun out there in out public API where is can (and likely will) be misused. 

Alternatively can existing "safe" APIs be used in tests instead of adding this function?

Reply:
I agree I don't like that it is `pub`. What do you think about having methods `insert_anchor` and `insert_seen_at` for Wallet? If not, we can try moving the tests that need it

Reply:
If this function is really only needed for tests can you leave it public but feature gate it with `#[cfg(test)]` ? Otherwise if `insert_anchor` and `insert_seen_at` can do their thing safely I like that approach, but again if they're only needed for testing I'd rather they not be in the public API.

Reply:
I think that means moving a lot of things back to unit tests including everything that uses `get_funded_wallet`. Should I do this in a separate PR?

Reply:
Can't you leave the tests were they are and just hide those new functions from the public API with the `#[cfg(test)]` ? I haven't tried it but I think it should work that way.

Reply:
I think it's okay to leave potentially-dangerous methods in wallet as long as it's documented. Potentially-dangerous also means powerful.

Reply:
Resolved by removing `insert_anchor` and `insert_seen_at`, using `apply_update` instead.